nnet_testing_script.m

```diff
@@ -0,0 +1,114 @@
+
+for k = 1:length(files)
+    %grab the next file
+    currFileName = files(k).name;
+    currFolder = files(k).folder;
+    currFileFullName = fullfile(currFolder, currFileName);
+
+    %grab the data for the current file
+    matData = load(currFileFullName);
+    
+    % for each image in the scan
+    for imageIdx = 1:numel(matData.adjusted_data_decembercal)
+        
+        image = matData.adjusted_data_decembercal(imageIdx);
+        currData = image.normalized_data;
+    
+        %extract features
+        features = extractFeatures(currData);
+    
+        %predict labels
+        labels = predict(model,features);
+        
+        insectDetected = any(labels);
+
+        %save label output
+        if insectDetected
+            insectDetectedStr = "insect";
+            insectExists = "true";
+        else
+            insectDetectedStr = "no-insect";
+        end
+
+        %array for storing rows where insects were detected
+        insect_rows = []
+        idx = 1
+        if insectDetected
+            for rowIdx = 1:size(labels)
+                if labels(rowIdx) == 1
+                    insect_rows(idx) = rowIdx
+                    idx = idx + 1
+                end
+            end    
+        end
+
+        imageFilenameTmp = split(replace(string(image.filename), "/", "-"), ".");
+        imageFilename = imageFilenameTmp(1); % remove the file extension
+        outprefix = join([modelName, imageFilename, insectDetectedStr], "-");
+
+        %use struct array to store file names, labels for images that
+        %contain insects
+        if insectDetected
+            nnetInsectImageFilesArray(a) = extractBetween(imageFilename, 1,15);
+            a = a + 1;
+            if i == 1
+                nnetInsectImages.fileName = imageFilename;
+                nnetInsectImages.labels = labels;
+                nnetInsectImages.isInsect = insectDetectedStr;
+                nnetInsectImages.insectRows = insect_rows;
+                i = 2;
+            else
+                nnetInsectImages(i).fileName = imageFilename;
+                nnetInsectImages(i).labels = labels;
+                nnetInsectImages(i).isInsect = insectDetectedStr;
+                nnetInsectImages(i).insectRows = insect_rows
+                i = i + 1;
+            end
+        end
+    end
+    
+end    
+%%
+temp = []
+%% 
+% Fill the temporary array with all the true unique images that contained insects
+
+for i = 1:23
+    temp = [temp; manual.insects(i).name];
+end
+temp2 = unique(temp, 'rows');
+%% 
+% Convert the string array that was created above into a cell array.
+
+nnetPredictedCMatrix = convertStringsToChars(nnetInsectImageFilesArray)
+%% 
+% Convert the cell array to a character vector, so that it is the same type 
+% as temp2.
+
+nnetPredictedCharArray = char(nnetPredictedCMatrix)
+%% 
+% Create a confusion matrix
+
+cMatrix = confusionchart(temp2, nnetPredictedCharArray)
```

Comment:
The primary issue is that we aren't using the right things for comparison here. We're using character arrays of the folder names that have predicted/"true" insects. We could probably use these character arrays to figure out which predicted folders also had manually labeled insects, but that won't tell us which images (the xxxx_PxxxTxxx.mat files inside each folder) had insects.

Reply:
@tvannoy Ahhh, I see. I seem to keep struggling to understand the discrepancies between the entire folder of images vs the individual images themselves. Thanks so much for your help!